### PR TITLE
wgpu: Implement Screen & Multiply blend modes as a trivial blends

### PR DESCRIPTION
**Pull request by Dinnerbone**
This is a 98% performance savings in rd_flying, and 37% in Heartcore

image
image

### CODE REVIEW

File: render/wgpu/src/blend.rs [hunk 1]
```diff
@@ -4,8 +4,6 @@ use swf::BlendMode;
 
 #[derive(Enum, Debug, Copy, Clone)]
 pub enum ComplexBlend {
-    Multiply,   // Can't be trivial, 0 alpha is special case
```
Comment (Aaron1011) [inline render/wgpu/src/blend.rs:7]: Were these comments incorrect?

Comment (Dinnerbone) [reply render/wgpu/src/blend.rs:7]: Kiiinda. 0 isn't specialcase, "not written yet" was. By making the default pixel color to white, it handles this case

**Comment by n0samu** [comment]
This is also a dramatic improvement in [Bejeweled 2](https://github.com/ruffle-rs/ruffle/files/10286637/bejeweled2.zip)! Tested in the desktop app on Windows 10 64-bit:
- ~65% GPU usage here vs. ~99% on master
- ~450MB of RAM vs. ~1GB on master
Very exciting! 🎉

**Comment by Lord-McSweeney** [comment]
I see a noticeable performance increase on rd_flying, but it still has noticeable lag.

**Comment by Lord-McSweeney** [comment]
Why are there different images uploaded for the tests? Shouldn't the output be the same?

**Comment by Dinnerbone** [comment]
> Why are there different images uploaded for the tests? Shouldn't the output be the same?

They are visually the same, but an unfortunate issue with image tests is that they rely on pixel by pixel similarities.

The way that hardware blends something vs the way a shader blends something is the same on paper, but some pixels around the edges of shapes where MSAA comes into play are subtly different. Neither one is more correct than the other, it's just how hardware graphics works.

**Review by Aaron1011 — APPROVED** [review]

**Pull request merged**